File: leetcode/26.remove-duplicates-from-sorted-array.cpp
/*
 * @lc app=leetcode id=26 lang=cpp
 *
 * [26] Remove Duplicates from Sorted Array
 */

// @lc code=start
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int i = 0, j = 1, k = 1;

        while (j<nums.size()) {
            if (nums[j]==nums[j-1]) {
                j++;
            } else {
                nums[i+1] = nums[j];
                j++;
                i++;
                k++;
            }
        }

        return k;
    }
};
// @lc code=end

File: leetcode/easy/26.remove-duplicates-from-sorted-array.py
#
# @lc app=leetcode id=26 lang=python3
#
# [26] Remove Duplicates from Sorted Array
#

# @lc code=start
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i, j, k = 0, 1, 1
        
        while j < len(nums):
            if nums[j] == nums[j-1]:
                j+=1
            else:
                nums[i+1] = nums[j]
                k+=1
                j+=1
                i+=1
        
        return k
            
            
            
            
# @lc code=end

